hail/python/test/hail/expr/test_expr.py

```diff
@@ -903,12 +903,17 @@ def test_aggregator_group_by_sorts_result(self):
 
     def test_agg_corr(self):
         ht = hl.utils.range_table(10)
-        ht = ht.annotate(x=hl.rand_unif(-10, 10),
-                         y=hl.rand_unif(-10, 10))
-        c, xs, ys = ht.aggregate((hl.agg.corr(ht.x, ht.y), hl.agg.collect(ht.x), hl.agg.collect(ht.y)))
+        ht = ht.annotate(tests=hl.range(10).map(
+            lambda i: hl.struct(
+                x=hl.cond(hl.rand_bool(0.1), hl.null(hl.tfloat64), hl.rand_unif(-10, 10)),
+                y=hl.cond(hl.rand_bool(0.1), hl.null(hl.tfloat64), hl.rand_unif(-10, 10)))))
 
-        scipy_corr, _ = pearsonr(xs, ys)
-        self.assertAlmostEqual(c, scipy_corr)
+        results = ht.aggregate(hl.agg.array_agg(lambda test: (hl.agg.corr(test.x, test.y), hl.agg.collect((test.x, test.y))), ht.tests))
+
+        for corr, xy in results:
+            filtered = [(x, y) for x, y in xy if x is not None and y is not None]
+            scipy_corr, _ = pearsonr([xy[0] for xy in filtered], [xy[1] for xy in filtered])
```

Comment:
I prefer this:

`pearsonr([x for x, _ in filtered], [y for _, y in filtered])`